cartography/intel/gcp/compute.py

```diff
@@ -0,0 +1,175 @@
+    :param http_error: The googleapi HttpError object
+    :return: The error reason as a string
+    """
+    try:
+        data = json.loads(http_error.content.decode('utf-8'))
+        if isinstance(data, dict):
+            reason = data['error']['errors'][0]['reason']
+        else:
+            reason = data[0]['error']['errors']['reason']
+    except Exception:
+        # Failed to parse the error. Let's fail the whole sync.
+        raise
+    return reason
+
+
+def get_zones_in_project(project_id, compute, max_results=None):
+    """
+    Return the zones where the Compute Engine API is enabled for the given project_id.
+    See https://cloud.google.com/compute/docs/reference/rest/v1/zones and
+    https://cloud.google.com/compute/docs/reference/rest/v1/zones/list.
+
+    If the API is not enabled, return None. The message logged in this case looks like this:
+    "WARNING:googleapiclient.http:Encountered 403 Forbidden with reason "accessNotConfigured"
+    INFO:googleapiclient.discovery:URL being requested:
+    GET https://www.googleapis.com/compute/v1/projects/<project_name>/zones?alt=json"
+    :param project_id: The project ID number to sync.  See  the `projectId` field in
+    https://cloud.google.com/resource-manager/reference/rest/v1/projects
+    :param compute: The compute resource object created by googleapiclient.discovery.build()
+    :param max_results: Optional cap on number of results returned by this function. Default = None, which means no cap.
+    :return: List of a project's zone objects if Compute API is turned on, else None.
+    """
+    try:
+        req = compute.zones().list(project=project_id, maxResults=max_results)
+        res = req.execute()
+        return res['items']
+    except HttpError as e:
+        reason = _get_error_reason(e)
+        if reason == 'accessNotConfigured':
+            logger.debug(
+                (
+                    "Google Compute Engine API access is not configured for project %s. "
+                    "Full details: %s"
+                ),
+                project_id,
+                e
+            )
+            return None
+        elif reason == 'notFound':
+            logger.debug(
+                (
+                    "Project %s returned a 404 not found error. "
+                    "Full details: %s"
+                ),
+                project_id,
+                e
+            )
+            return None
+        elif reason == 'forbidden':
+            # Write some additional advice and then re-raise
+            logger.error(
+                (
+                    "Encountered a 403-Forbidden error trying to list zones on project %s - Please ensure that the "
+                    "identity that your GOOGLE_APPLICATION_CREDENTIALS point to has the securityReviewer role defined "
+                    "on this project."
+                ),
+                project_id
+            )
+            raise
```

Comment:
Because we log an ERROR-level message for all uncaught exceptions in `Sync.run` this block will result in multiple log messages for the same error. I would suggest removing this branch entirely and letting the exception be bubble up from the else branch or, if you feel the extra context is necessary, create a new exception object, wrap the HTTPError, and raise the new exception.